### PR TITLE
Feature/12931 1

### PR DESCRIPTION
**Pull request by shabeeb-aot**
*Issue #:* /bcgov/entity#12931

*Description of changes:*
* unit test case 
* some component conditions updated to show in special resolution  

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/App.vue
Comment (severinbeauvais) [inline src/App.vue:352]: Nice cleanup 👍

File: src/components/common/YourCompany/YourCompany.vue
Comment (severinbeauvais) [inline src/components/common/YourCompany/YourCompany.vue:187]: Could use some cleanup here, or comment updates.

File: src/store/getters/state-getters.ts
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts:425]: Also update comments re: alteration.

File: src/views/SpecialResolution.vue
Comment (severinbeauvais) [inline src/views/SpecialResolution.vue:40]: You're using the AlterationSummary component. Does it work for you or do you need your own? If you're reusing it, please look into renaming it (and moving its location from alteration-only to common or something).

File: tests/unit/SpecialResolution.spec.ts
Comment (severinbeauvais) [inline tests/unit/SpecialResolution.spec.ts:29]: You may not need the KC token for your tests. Usually it's App.vue that needs it. Check and clean up if not needed.

File: tests/unit/SpecialResolution.spec.ts
Comment (severinbeauvais) [inline tests/unit/SpecialResolution.spec.ts:79]: Does special resolution support future effective? If not, please clean up.

File: tests/unit/SpecialResolution.spec.ts
Comment (severinbeauvais) [inline tests/unit/SpecialResolution.spec.ts:108]: Does special resolution need name translation? If not, please clean up.

Same for other mocks here.

File: tests/unit/SpecialResolution.spec.ts
Comment (severinbeauvais) [inline tests/unit/SpecialResolution.spec.ts:366]: I know that we've left some other unit tests incomplete, but if you have time then it would be great to finish this unit tests, since it will verify loading data from a draft versus the business snapshot, which is important functionality.

File: src/views/SpecialResolution.vue
Comment (shabeeb-aot) [reply src/views/SpecialResolution.vue:351]: I was getting some errors in the codebase. so changed to public. I think this will may be outside the class, What would you prefer?
Screen Shot 2022-07-25 at 12 55 04 PM

File: src/components/SpecialResolution/SpecialResolutionSummary.vue
Comment (severinbeauvais) [inline src/components/SpecialResolution/SpecialResolutionSummary.vue:245]: You don't need this if future effective Special Resolutions aren't supported.

Maybe you don't need alterationDateTimeInvalid() or some template components either.

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:700]: Typo in name.

Also, you need a buildSpecialResolutionFiling, right?

File: tests/unit/SpecialResolution.spec.ts
Comment (shabeeb-aot) [reply tests/unit/SpecialResolution.spec.ts:29]: without KC token , the test is not passing

File: tests/unit/SpecialResolution.spec.ts
Comment (shabeeb-aot) [reply tests/unit/SpecialResolution.spec.ts:327]: I will add more test case when we are working on components

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:736]: You don't need this for CPs.

I'd prefer to not review your PR each time you add or clean up functionality. I agree with you that small PRs are nice but if we merged this code then there would be tech debt to clean it up later and I don't know how we'd keep track of that, so I think it's worth some effort up front to only merge changes that won't need cleanup later.

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:729]: More cleanup in the comments at least, and maybe functionality depending on what's needed in a Special Resolution filing.

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [reply src/mixins/filing-template-mixin.ts:736]: ^^ less tech debt not more, please

File: src/views/SpecialResolution.vue
Comment (severinbeauvais) [reply src/views/SpecialResolution.vue:351]: In my opinion, the template is part of this class (even though your TS is complaining).

Do we know what the future holds for property/method visibility? For example, if visibility will be enforced soon, and the template is not part of the class, then sadly this method will need to be public (but I feel strongly that it really shouldn't be exposed).

I get visibility warnings from Vetur, not TS. Should I be using your VS Code extensions, or should you be using mine?

File: tests/unit/SpecialResolution.spec.ts
Comment (severinbeauvais) [reply tests/unit/SpecialResolution.spec.ts:29]: OK. Maybe it's looking at the "is staff" or "is premium" getters, which I think look at the role from the token.

In your opinion, would it be cleaner to mock the getters to return True according to your test needs, or is it OK to leave the sample KC token here?

File: tests/unit/SpecialResolution.spec.ts
Comment (severinbeauvais) [reply tests/unit/SpecialResolution.spec.ts]: I think you still have other mocked API calls that you don't need.

File: src/views/SpecialResolution.vue
Comment (shabeeb-aot) [reply src/views/SpecialResolution.vue:351]: yeah, I think its coming from vuter Extention

File: tests/unit/SpecialResolution.spec.ts
Comment (shabeeb-aot) [reply tests/unit/SpecialResolution.spec.ts:29]: Looks like its checking only if the value exists, when I am giving any value, its working, not on blank, I will give a sample value (not token) (in all other filing we are using the same token) , if you are good

File: src/mixins/filing-template-mixin.ts
Comment (shabeeb-aot) [reply src/mixins/filing-template-mixin.ts]: yeah agree, Since this is a basic setup, I tried to keep most of the components there. WE have tickets to work on each component. This part will come on click on save and resume I suppose. I removed most of the component, i feel like not using / not going to use in future

File: src/mixins/filing-template-mixin.ts
Comment (shabeeb-aot) [inline src/mixins/filing-template-mixin.ts:310]: Not using this now, will implement functionality as a part of save ticket

File: src/views/SpecialResolution.vue
Comment (severinbeauvais) [reply src/views/SpecialResolution.vue:351]: So, do you think it should be protected or public?

(I don't like the idea of this internal method being public.)

Comment (shabeeb-aot) [reply src/views/SpecialResolution.vue:351]: I will update with protected

File: src/components/SpecialResolution/SpecialResolutionSummary.vue
Comment (severinbeauvais) [inline src/components/SpecialResolution/SpecialResolutionSummary.vue:84]: Not needed?

File: src/components/SpecialResolution/SpecialResolutionSummary.vue
Comment (severinbeauvais) [inline src/components/SpecialResolution/SpecialResolutionSummary.vue:97]: Not needed?

File: src/components/SpecialResolution/SpecialResolutionSummary.vue
Comment (severinbeauvais) [inline src/components/SpecialResolution/SpecialResolutionSummary.vue:115]: Not needed?

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:347]: Not needed?

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:342]: Not needed?

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:777]: Not needed?

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:816]: Not needed?

**Review by severinbeauvais — APPROVED** [review]
LGTM (after a bit of cleanup per outstanding comments).

File: src/mixins/filing-template-mixin.ts
Comment (shabeeb-aot) [reply src/mixins/filing-template-mixin.ts:342]: we need this since we are using a name request component, COOP can change a name using name request, correct me i am wrong

File: src/mixins/filing-template-mixin.ts
Comment (shabeeb-aot) [reply src/mixins/filing-template-mixin.ts:768]: we need this since we are using a name request component, COOP can change a name using name request, correct me I am wrong, I think this is to set namerequest value

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [reply src/mixins/filing-template-mixin.ts:342]: Keep it if you need it. I don't know if coops can change their name.

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [reply src/mixins/filing-template-mixin.ts:768]: This is to restore NR data when loading a draft filing.

Again, keep this if you need but I don't know.

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:804]: You will need to store the Staff Payment data from a draft filing (if staff can complete this filing).

**Review by severinbeauvais — APPROVED** [review]
Looks good to me.

You can merge after your changes (if they are small) and after another reviewer or 2 have approved.

File: src/mixins/filing-template-mixin.ts
Comment (shabeeb-aot) [reply src/mixins/filing-template-mixin.ts:342]: yeah, As per design coop can change their name

Screen Shot 2022-07-26 at 11 31 09 AM

File: src/mixins/filing-template-mixin.ts
Comment (shabeeb-aot) [reply src/mixins/filing-template-mixin.ts:804]: As far as design, there is no staff version now. I hope it will come soon, So i removed it now and when we have design we can add it back

**Review by lewischenstudio — APPROVED** [review]

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [reply src/mixins/filing-template-mixin.ts:342]: OK, FYI: Lewis has changed some things in the name request mixin that you will need to update. Specifically, you will need to state which NR types are valid for your scenario.

Ref: https://github.com/bcgov/business-edit-ui/pull/333

File: src/mixins/filing-template-mixin.ts
Comment (severinbeauvais) [reply src/mixins/filing-template-mixin.ts:804]: Sounds good.

And you can delete `buildStaffPayment` in the build method so it doesn't set values incorrectly.

File: src/mixins/filing-template-mixin.ts
Comment (shabeeb-aot) [reply src/mixins/filing-template-mixin.ts:342]: Good to know, I will add reference in name change ticket https://app.zenhub.com/workspaces/relationships-team-space-61435c47e483c4000f08e9f6/issues/bcgov/entity/12894

**Comment by shabeeb-aot** [comment]
can I update version and merge if you are good?

**Comment by severinbeauvais** [comment]
> can I update version and merge if you are good?

Yes, I see you have another approval 👍

**Pull request merged**
